collections/app/controllers/CollectionsController.scala

```diff
@@ -126,11 +137,35 @@ object CollectionsController extends Controller with ArgoHelpers {
     } getOrElse Future.successful(invalidJson(req.body))
   }
 
+  type MaybeTree = Option[Node[Collection]]
+  def hasChildren(path: List[String]): Future[Boolean] =
+    allCollections.map { tree =>
```

Comment:
I think the structure of the data store prevents this from being efficient. ccing @jamesgorrie to see if he thinks different.